src/buildercore/bootstrap.py

```diff
@@ -51,7 +51,8 @@ def escape_string_parameter(parameter):
 
     env_string = ['%s=%s' % (k, v) for k, v in environment_variables.items()]
     cmd = ["/bin/bash", remote_script] + lmap(escape_string_parameter, list(script_params))
-    retval = remote_sudo(" ".join(env_string + cmd))
+    result = remote_sudo(" ".join(env_string + cmd))
+    retval = result['return_code']
```

Comment:
was this wrong while using the original Fabric `remote_sudo`, too? As it seems the backend can be switched from Fabric to Threadbare and back, I assume this code supports both

Reply:
It wasn't wrong, I've made [the return value consistent](https://github.com/elifesciences/builder/pull/586/files/155b9bbbc6129889bbdbfa98fa4c4f1c63badab7#diff-18a6b93c14cdde804a9939c596c81067R62-R69) across backends. The Fabric backend now returns a dictionary of results for `remote`, `local` and `remote_sudo` versus just the return code with stdout, stderr etc attached as attributes to an object.